sdk/rust/src/entity_id.rs

```diff
@@ -67,6 +67,16 @@ impl Debug for Checksum {
     }
 }
 
+pub trait AutoValidateChecksum {
+    fn validate_checksum_for_ledger_id(&self, ledger_id: &LedgerId) -> Result<(), Error>;
+}
+
+impl<ID> AutoValidateChecksum for Option<ID> where ID: AutoValidateChecksum {
+    fn validate_checksum_for_ledger_id(&self, ledger_id: &LedgerId) -> Result<(), Error> {
+        self.as_ref().map_or(Ok(()), |id| id.validate_checksum_for_ledger_id(ledger_id))
```

Comment:
this is actually one of those times where I think... hmm
```
if let Some(id) = self {
    return // ... 
}

Ok(())
```
might be more understandable